metadata-ingestion/src/datahub/ingestion/source/s3/source.py

```diff
@@ -414,11 +413,38 @@ def get_fields(self, table_data: TableData, path_spec: PathSpec) -> List:
                 fields = json.JsonInferrer().infer_schema(file)
             elif extension == ".avro":
                 fields = avro.AvroInferrer().infer_schema(file)
+            elif extension == "":
+                try:
+                    fields = json.JsonInferrer().infer_schema(file)
+                except Exception as e:
```

Comment:
Let's only catch exceptions pertaining to inferring of the schema. We want to raise an exception in other cases, such as 'file not found', etc.